### PR TITLE
SDI-543 Delete key date adjustment

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingAdjustmentResource.kt [hunk 1]
```diff
@@ -112,16 +112,6 @@ class SentencingAdjustmentResource(private val sentencingAdjustmentService: Sent
           )
         ]
       ),
-      ApiResponse(
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingAdjustmentResource.kt:115]: we don't return a 404 :-)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentenceAdjustmentsResourceIntTest.kt [hunk 2]
```diff
@@ -738,7 +738,6 @@ class SentenceAdjustmentsResourceIntTest : IntegrationTestBase() {
         .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_SENTENCING")))
         .exchange()
         .expectStatus().isNotFound
-        .expectBody()
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentenceAdjustmentsResourceIntTest.kt:741]: unnecessary

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingAdjustmentResource.kt [hunk 3]
```diff
@@ -338,6 +328,45 @@ class SentencingAdjustmentResource(private val sentencingAdjustmentService: Sent
...
+  @PreAuthorize("hasRole('ROLE_NOMIS_SENTENCING')")
+  @ResponseStatus(HttpStatus.NO_CONTENT)
+  @DeleteMapping("/key-date-adjustments/{adjustmentId}")
+  @Operation(
+    summary = "deletes specific key date adjustment",
+    description = "Requires role NOMIS_SENTENCING. Deletes a key date adjustment by id",
+    responses = [
+      ApiResponse(
+        responseCode = "204",
+        description = "the key date adjustment has been deleted"
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingAdjustmentResource.kt:340]: probably worth adding that it has been deleted or it was not found so no deletion took place

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**